File: 프로그래머스/3/12938. 최고의 집합/최고의 집합.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int n, int s) {
    vector<int> answer;

    int base = s / n;
    int rem = s % n;

    if (base == 0) return {-1};

    for (int i = 0; i < n - rem; i++) 
        answer.push_back(base);
        
    for (int i = 0; i < rem; i++) 
        answer.push_back(base + 1);

    return answer;
}

File: 프로그래머스/3/12938. 최고의 집합/최고의 집합.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public int[] solution(int n, int s) {
        if (s < n) return new int[]{-1}; 
        
        int base = s / n;
        int rem = s % n;
        
        int[] answer = new int[n];
        Arrays.fill(answer, base);
        
        for (int i = n - 1; i >= n - rem; i--) {
            answer[i]++;
        }
        
        return answer;
    }
}